src/cargo/ops/cargo_clean.rs

```diff
@@ -104,7 +105,7 @@ pub fn clean(ws: &Workspace<'_>, opts: &CleanOptions<'_>) -> CargoResult<()> {
     // Doc tests produce no output.
 
     // Get Packages for the specified specs.
-    let mut pkg_ids = Vec::new();
+    let mut pkg_ids = BTreeSet::new();
```

Comment:
Can you add a test for this?  Ideally the first commit would have the test with a `#[should_panic]` and then the second commit would include this with the test updated.   Using these two commits will make it easier to ensure the test is testing the right thing and to show to reviewers what the intended behavior change is (especially since this is a PR without an issue)

Reply:
I took a quick look at this, and adding a test seems fairly complicated. There are not any tests in cargo_clean.rs, and most of it consists of a monolithic clean() function, in which this code change is buried. I would need to construct an appropriate Workspace object, which is fairly complicated.

It looks like there are some cursory integration tests, but adding something would require figuring out exactly how to trigger a duplicate ID, which might be tricky.

Overall, it doesn't look at all straightforward to test. :( I will keep poking at it, though, and see if I can figure anything out.

Reply:
Look in `tests/testsuite/clean.rs` for tests.